packages/bits/src/lib/wizard-v2/types.ts

```diff
@@ -2,3 +2,21 @@
 export interface IWizardState {
     finished: boolean;
 }
+
+export interface IWizardConfig {
+    stepStateIcons: IWizardStepStateIconConfig;
+    [key: string]: any;
+}
+
+export interface IWizardStepStateIconConfig {
+    icons?: Partial<IWizardIconStates>;
+    colors?: Partial<IWizardIconStates>;
+}
```

Comment:
I feel like we should couple the icons with the colors so that each icon state is self-contained instead of having separated properties.

```
export interface IWizardStepStateConfig {
    iconsConfig?: Partial<IWizardStateMap<IIconConfig>>;
}

export interface IWizardStateMap<T> {
    active: T;
    visited: T;
    initial: T;
    error: T;
    [key: string]: T;
}

export interface IIconConfig {
    icon: string;
    color: string;
} 
```

Reply:
It depends on how we look at what we want to configure there. What you suggest looks more like configuring the step state first and then specify what exactly to configure inside the step state configuration, while in my case we configure the icons config first and don't bother with the rest. I like to keep things separated, but I agree that my way is a bit more specific. On the other hand, with this approach, it is a bit easier to only set the icons or colors separately, and it seems to be a bit more obvious from the consumer's point of view: in case they're only interested in configuring state icons they just send the appropriate config.

Also, there are additional questions that arise when attempting to configure the step state in the way you suggest: do we configure it for the step, or for the header? Icons only make sense for the header, but their type depends on the step state information, so how should we set this config: for the step, or for the header? In case we choose neither of these and simply add it as a root config option then what's the difference from what's already done, but specifically for the icons? Moreover, the step state does not know anything about any icons, it is just a part of the step.

If we really want to think everything through and make the interface for the wizard config as expandable as possible, we should probably consider starting from this structure:

`
{
    .... <wizard global settings>
    step:
        header?: IWizardHeaderConfig;
        body?: IWizardBodyConfig;
        footer?: IWizardFooterConfig;
    .... <other configs>
}
`
And each of these interfaces will (or will not) contain other fields and their subsequent structures, which maybe will never be used by anyone ever.

That's the way of thinking I had. We have no idea what consumers will want to configure in the future, and it seemed logical to me to keep it simple and start from the minimum needed configuration structure for the wizard and later expand the interface by simply adding new optional configuration entries, which is never a breaking change. Later on, in case we have so many configuration options that they'll start to pile up, we'll want to segregate the interface, and we'll do that in a major version bump.

What's your opinion on this @lukasholcik, @Blakesalot ?